apps/src/templates/SchoolDataInputs.jsx

```diff
@@ -1,99 +1,138 @@
 import {COUNTRIES} from '@cdo/apps/geographyConstants';
 
-const TEST_SCHOOL_DATA = [{value: 'test', text: 'VeryLongNameTestSchool'}];
 export default function SchoolDataInputs({
-  onCountryChange,
-  onSchoolChange,
-  onSchoolNotFoundChange,
-  country = '',
-  schoolType = '',
-  ncesSchoolId = '',
-  schoolName = '',
-  schoolCity = '',
-  schoolState = '',
-  schoolZip = '',
-  schoolLocation = '',
-  useLocationSearch = false,
+  includeHeaders = true,
   fieldNames = {
-    schoolType: 'user[school_info_attributes][school_type]',
     country: 'user[school_info_attributes][country]',
     ncesSchoolId: 'user[school_info_attributes][school_id]',
     schoolName: 'user[school_info_attributes][school_name]',
-    schoolState: 'user[school_info_attributes][school_state]',
     schoolZip: 'user[school_info_attributes][school_zip]',
-    googleLocation: 'user[school_info_attributes][full_address]',
   },
-  showErrors,
-  showRequiredIndicator,
 }) {
-  let COUNTRY_ITEMS = [];
+  const [askForZip, setAskForZip] = useState(false);
+  const [isOutsideUS, setIsOutsideUS] = useState(false);
+  const [zip, setZip] = useState('');
+  const [country, setCountry] = useState('');
+  const [schoolName, setSchoolName] = useState('');
+  const [zipSearchReady, setZipSearchReady] = useState(false);
+
+  let COUNTRY_ITEMS = [{value: 'selectCountry', text: i18n.selectCountry()}];
   for (const item of Object.values(COUNTRIES)) {
     COUNTRY_ITEMS.push({value: item.label, text: item.value});
   }
 
+  useEffect(() => {
+    if (zip.length === 5) {
+      setZipSearchReady(true);
+    } else {
+      // Removes the school dropdown if you delete part of the zip
+      setZipSearchReady(false);
+    }
+  }, [zip.length]);
+
+  const onCountryChange = e => {
+    const country = e.target.value;
+    setCountry(country);
+    // We don't want to display any fields to start that won't eventually be
+    // necessary, so updating both of these any time country changes
+    if (country === 'US') {
+      setAskForZip(true);
+      setIsOutsideUS(false);
+    } else {
+      setAskForZip(false);
+      setIsOutsideUS(true);
+    }
+  };
+
   return (
     <div className={style.outerContainer}>
-      <Heading2 className={style.topPadding}>{i18n.censusHeading()}</Heading2>
-      <BodyTwoText>{i18n.schoolInfoInterstitialTitle()}</BodyTwoText>
+      {includeHeaders && (
+        <div>
+          <Heading2 className={style.topPadding}>
+            {i18n.censusHeading()}
+          </Heading2>
+          <BodyTwoText>{i18n.schoolInfoInterstitialTitle()}</BodyTwoText>
+        </div>
+      )}
       <div className={style.inputContainer}>
         <BodyTwoText className={style.padding} visualAppearance={'heading-xs'}>
           {i18n.whatCountry()}
         </BodyTwoText>
         <SimpleDropdown
           className={style.dropdown}
+          name={fieldNames.country}
           items={COUNTRY_ITEMS}
-          name="countryDropdown"
```

Comment:
this might be out of scope for this PR, but do you need to pass down the form name field here? ie one of the values defined [here](https://github.com/code-dot-org/code-dot-org/blob/03b428298e8b2bcfdc8bf90ff70ac4dadc3adac1/apps/src/templates/SchoolInfoInputs.jsx#L53)

Reply:
jk never mind sorry I completely missed the fact that this line moved 🤦‍♀️ 🤦‍♀️